operator/src/test/java/org/keycloak/operator/OperatorE2EIT.java

```diff
@@ -0,0 +1,75 @@
+package org.keycloak.operator;
+
+import io.fabric8.kubernetes.api.model.Node;
+import io.fabric8.kubernetes.client.KubernetesClientException;
+import io.fabric8.kubernetes.client.dsl.Resource;
+import io.quarkus.test.junit.QuarkusTest;
+import org.awaitility.Awaitility;
+import org.awaitility.core.ConditionTimeoutException;
+import org.junit.jupiter.api.Test;
+import org.keycloak.operator.v2alpha1.crds.Keycloak;
+
+import java.time.Duration;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@QuarkusTest
+public class OperatorE2EIT extends ClusterOperatorTest {
+
+    @Test
+    public void given_ClusterAndOperatorRunning_when_KeycloakCRCreated_Then_KeycloakStructureIsDeployedAndStatusIsOK() {
+        logger.info(((testremote) ? "Remote " : "Local ") + "Run Test :" + operator + " -- " + namespace);
+
+        // Node
+        List<Node> nodes = k8sclient.nodes().list().getItems();
+        assertThat(nodes).hasSize(1);
+
+        // NS created by the extension [ probably this doesnt make sense as if not passes, then extension would fail]
+        Awaitility.await()
+                .atMost(Duration.ofSeconds(5))
+                .pollDelay(Duration.ofSeconds(1))
+                .untilAsserted(() -> assertThat(k8sclient.namespaces().withName(namespace).get()).isNotNull());
+        assertThat(k8sclient.namespaces().withName(namespace + "XX").get()).isNull();
+
+        // CR
+        Resource<Keycloak> keycloakResource = k8sclient.resources(Keycloak.class).load("kubernetes/example-keycloak.yml");
+        k8sclient.resources(Keycloak.class).inNamespace(namespace).createOrReplace(keycloakResource.get());
+
+        Awaitility.await()
+                .atMost(Duration.ofSeconds(5))
+                .pollDelay(Duration.ofSeconds(1))
+                .untilAsserted(() -> assertThat(k8sclient.resources(Keycloak.class).inNamespace(namespace).withName("example-kc").get()).isNotNull());
```

Comment:
I think we don't need to test fabric8 `kubernetes-client`, the assertion below is totally enough for me.

Reply:
well, in fact all the tests are redundant except the last one if you think that.... but every check is providing value
but anyway, ok